application.go

```diff
@@ -431,12 +429,14 @@ func (r *marathonClient) WaitOnApplication(name string, timeout time.Duration) e
 			flick.SwitchOn()
 		}()
 		for !flick.IsSwitched() {
-			if found, err := r.HasApplication(name); err != nil {
-				continue
-			} else if found {
-				if app, err := r.Application(name); err == nil && app.AllTaskRunning() {
-					return nil
+			if err := r.HasApplication(name); err != nil {
```

Comment:
@eicca makes sense ... since you can rely on the custom error from Application() ..